js/adapt-contrib-resources.js

```diff
@@ -46,9 +46,28 @@ class Resources extends Backbone.Controller {
         resources = resources.filter(resource => resource._isGlobal !== false);
       }
       model.set('_resources', resources);
+
+      this.setupTypes(model, resourcesData);
+
       drawer.triggerCustomView(new ResourcesView({ model }).$el);
     });
   }
+
+  setupTypes(model, resourcesData) {
+    const types = ['all']; // must contain 'all'
+    const filterButtons = resourcesData._filterButtons;
+
+    for (const key in filterButtons) {
+      const typeName = key;
+
+      if (typeName !== 'all') {
+        types.push(typeName);
+      }
+    };
+
+    model.set('_resourceTypes', types);
+  }
```

Comment:
```suggestion
  setupTypes(model, resourcesData) {
    const configuredTypes = Object.keys(resourcesData._filterButtons).filter(type=> type !== 'all');
    const allTypes = [ 'all', ...configuredTypes ];
    model.set('_resourceTypes', allTypes);
  }
```

```js

```
Rather that iterating, try using these:
[`Object.keys`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys),[`Object.values`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values),[`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)
[`... spread syntax`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_array_literals)

The oddity of what's actually happening here is a little clearer when the main focus isn't a loop.
1. Get keys from configured buttons
2. Filter to remove `all`
3. Add `all` to the front
4. Set on the model

Otherwise a `continue` early, similar to `return` early works to reduce the nesting:
```js
    for (const typeName in filterButtons) {
      if (typeName === 'all') continue;
      types.push(typeName);
    }
```